diff --git a/src/nez/ext/Gtpeg.java b/src/nez/ext/Gtpeg.java
index 55ee2f3..0c5e24d 100644
--- a/src/nez/ext/Gtpeg.java
+++ b/src/nez/ext/Gtpeg.java
@@ -51,7 +51,7 @@ public class Gtpeg extends GrammarFileLoader {
 	@Override
 	public void parse(Tree<?> node) {
 		RootExpr rootExpr = (RootExpr) this.translate(node);
-		TypeEnv env = new TypeEnv();
+		TypeEnv env = TypeEnv.getInstance();
 		new ExprTypeChecker(env).visit(rootExpr);
 		new NezGrammarGenerator().visit(rootExpr, this.getGrammar());
 	}
